RealtimeAppBackend/RealtimeApp/views.py

```diff
@@ -43,21 +43,55 @@ def retrieveData(request):
 @csrf_exempt
 def retrieveBluetooth(request):
     # @TODO(matthew-ardi): Please put the code where you receive the data from bluetooth signal here and
-    s = serial.Serial('/dev/ttyS0', 115200, timeout=1)
+    # @NOTE(sonminhtran1997): this might not work when running server on your side because it needs our (matthew-ardi) device
+    ##
+    ##  Getting a set of data through serial communication of the device bluetooth
+    ##
+    s = serial.Serial('/dev/ttyS0', 115200, timeout=1) 
     string = ""
     final_message = ""
-    while(1):
-        c = s.read()    # wait for start character
-        if c == '#':
-            for i in range (0,7):
-                c = s.readline()
-                string += c
-            break
-    
-    final_message = str(string.split('#', 1)[1])    # String of data transmitted through bluetooth from device
+    try:
+        while(1):
+            c = s.read()    # wait for start character
+            if c == '#':
+                for i in range (0,7):
+                    c = s.readline()
+                    string += c
+                break
+        
+        final_message = str(string.split('#', 1)[1])    # String of data transmitted through bluetooth from device
+    except:
+        return  # sometime exception might happen when server is trying to read serial data but no new data has been sent. 
+                # when this happen, skip the process below and let the server re-request serial data.
+
+    ##
+    ##  splitting messages to read boolean values of each event detection
+    ##
+    split_message = final_message.split('\n')
+    roomLightState = False
+    benchLightState = False
+    blenderState = False
+    spaceHeaterState = False
+    roomOccupiedState = False
+    grinderState = False
+
+    event_state = []
+    for event in split_message:
+        state = event.split(' ')
+        if "true" in state:
+            event_state.append(True)
+        else:
+            event_state.append(False)
+
+    roomLightState = event_state[0]
+    benchLightState = event_state[1]
+    blenderState = event_state[2]
+    spaceHeaterState = event_state[3]
+    roomOccupiedState = event_state[4]
+    grinderState = event_state[5]
 
     # @TODO:               then modify the global object data
-
+    data(roomLightState, benchLightState, blenderState, spaceHeaterState, roomOccupiedState, grinderState)
```

Comment:
@sonminhtran1997 not sure if this is the right sytax. please fix it if the syntax is wrong. thank you.

Reply:
for this there is a function setData at the end of this file. Basically you just need to do this
setData(data, roomLightState, benchLightState, blenderState, spaceHeaterState, roomOccupiedState, grinderState)

Please help me update this function